app/scripts/controllers/transactions/index.js

```diff
@@ -264,7 +264,12 @@ class TransactionController extends EventEmitter {
       // must set transaction to submitted/failed before releasing lock
       nonceLock.releaseLock()
     } catch (err) {
-      this.txStateManager.setTxStatusFailed(txId, err)
+      // this is try-catch wrapped so that we can guarantee that the nonceLock is released
+      try {
+        this.txStateManager.setTxStatusFailed(txId, err)
```

Comment:
Was `setTxStatusFailed()` throwing uncaught errors? Looking at it, it seems so straight forward. In fact, even its own internal `_setTxStatus(id, status)` method has its own try/catch.

Reply:
i treated it as a black box and assumed errors could come from it
the errors could even be inside a eventemitter handler